tests/e2e/datadog/metrics/series.rs

```diff
@@ -0,0 +1,275 @@
+
+    in_payloads.iter().for_each(|payload| {
+        let series = payload.data.as_array().unwrap();
+
+        series.iter().for_each(|serie| {
+            let ser: DatadogSeriesMetric = serde_json::from_value(serie.clone()).unwrap();
+            out_series.push(ser);
+        });
+    });
+
+    out_series
+}
+
+async fn get_v1_series_from_pipeline(address: String) -> SeriesIntake {
+    info!("getting v1 series payloads");
+    let payloads =
+        get_fakeintake_payloads::<FakeIntakeResponseJson>(&address, SERIES_ENDPOINT_V1).await;
+
+    info!("unpacking payloads");
+    let payloads = unpack_v1_series(&payloads.payloads);
+    info!("converting payloads");
+    let payloads = convert_v1_payloads_v2(&payloads);
+
+    info!("generating series intake");
+    let intake = generate_series_intake(&payloads);
+
+    common_series_assertions(&intake);
+
+    info!("{:?}", intake);
+
+    intake
+}
+
+async fn get_v2_series_from_pipeline(address: String) -> SeriesIntake {
+    info!("getting v2 series payloads");
+    let payloads =
+        get_fakeintake_payloads::<FakeIntakeResponseRaw>(&address, SERIES_ENDPOINT_V2).await;
+
+    info!("unpacking payloads");
+    let payloads = unpack_proto_payloads::<MetricPayload>(&payloads);
+
+    info!("generating series intake");
+    let intake = generate_series_intake(&payloads);
+
+    common_series_assertions(&intake);
+
+    info!("{:?}", intake);
+
+    intake
+}
+
+pub(super) async fn validate() {
+    info!("==== getting series data from agent-only pipeline ==== ");
+    let agent_intake = get_v2_series_from_pipeline(fake_intake_agent_address()).await;
+
+    info!("==== getting series data from agent-vector pipeline ====");
+    let vector_intake = get_v1_series_from_pipeline(fake_intake_vector_address()).await;
+
+    assert_eq!(
+        agent_intake.len(),
+        vector_intake.len(),
+        "different number of unique Series contexts received"
+    );
+
+    agent_intake
+        .iter()
+        .zip(vector_intake.iter())
+        .for_each(|(agent_ts, vector_ts)| {
+            assert_eq!(agent_ts.0, vector_ts.0, "Mismatch of series context");
+
+            let metric_type = agent_ts.0.r#type;
+
+            // gauge: last one wins.
+            // we can't rely on comparing each value due to the fact that we can't guarantee consistent sampling
+            if metric_type == 3 {
+                let last_agent_point = agent_ts.1.iter().last();
+                let last_vector_point = vector_ts.1.iter().last();
+
+                assert_eq!(
+                    last_agent_point, last_vector_point,
+                    "Mismatch of gauge data"
+                );
+            }
+
+            // rate: summation.
+            if metric_type == 2 {
+                let mut agent_sum = 0.0;
+                agent_ts
+                    .1
+                    .iter()
+                    .for_each(|(_tb, points)| points.iter().for_each(|v| agent_sum += v));
```

Comment:
nit: Iterators have a `sum` adapter:
```suggestion
                let agent_sum = agent_ts
                    .1
                    .iter()
                    .map(|(_tb, points)| points.iter().sum())
                    .sum();
```